keps/sig-storage/3751-volume-provisioned-io/README.md

```diff
@@ -0,0 +1,1338 @@
+
+
+###### What specific metrics should inform a rollback?
+
+A metric `controller_modify_volume_errors_total` will indicate a problem with the feature.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Upgrade and rollback will be tested when the feature gate will change to beta.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+A counter metric will be present in external-resizer metric endpoint, it will show total
+count of successful and failed ControllerModifyVolume.
+
+###### How can someone using this feature know that it is working for their instance?
+
+By inspecting a `controller_modify_volume_total` metric value. If the counter
+is increasing while letting PVCs being updated retroactively the feature is enabled. And at the same time if
+`controller_modify_volume_total` counter does not increase the feature
+works as expected.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+<!--
+This is your opportunity to define what "normal" quality of service looks like
+for a feature.
+
+It's impossible to provide comprehensive guidance, but at the very
+high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99.9% of /health requests per day finish with 200 code
+
+These goals will help you determine what you need to measure (SLIs) in the next
+question.
+-->
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [ ] Metrics
+    - Metric name: `controller_modify_volume_errors_total` and `controller_modify_volume_errors_total`
+    - [Optional] Aggregation method:
+    - Components exposing the metric: external-resizer
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+<!--
+Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+implementation difficulties, etc.).
+-->
+
+### Dependencies
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### Does this feature depend on any specific services running in the cluster?
+
+external-resizer.
+
+### Scalability
+
+<!--
+For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them.
+
+For beta, this section is required: reviewers must answer these questions.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### Will enabling / using this feature result in any new API calls?
```

Comment:
The VAC protection controller will be expensive because it needs to LIST PVCs and maybe PVs